### PR TITLE
Fix camera flying away when pitching

### PR DESCRIPTION
**Pull request by kiryldz**
### Summary of changes

For the shove gesture (changing the pitch) we should always explicitly use screen center as an anchor point.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.

| Before | After |
| ----- | ----- |
| <video src="https://user-images.githubusercontent.com/15800566/182842886-9faaaf3a-cde9-4e8d-8798-d5992d1fa544.mp4" width = 250/> | <video src="https://user-images.githubusercontent.com/15800566/182842976-44399725-abff-4051-add6-478da58b6159.mp4" width = 250/> |
        
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Review by pengdev — APPROVED** [review]
Great catch!

**Review by alexander-kulikovskii — APPROVED** [review]

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 1]
```diff
@@ -1047,8 +1048,9 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
       pitch - (SHOVE_PIXEL_CHANGE_FACTOR * (deltaPixelsSinceLast + deferredShove))
     deferredShove = 0.0
     pitch = clamp(optimizedPitch, MINIMUM_PITCH, MAXIMUM_PITCH)
+    // changing pitch always should happen relative to the screen center otherwise camera may fly away
     easeToImmediately(
-      CameraOptions.Builder().pitch(pitch).build(),
+      CameraOptions.Builder().anchor(centerScreen).pitch(pitch).build(),
```
Comment (yunikkk) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1053]: Isn't it reproduced for other gestures?

File: plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt [hunk 2]
```diff
@@ -852,12 +853,18 @@ class GesturesPluginTest {
...
     )
+    val centerScreen = ScreenCoordinate(50.0, 50.0)
+    presenter.centerScreen = centerScreen
     val listener: OnShoveListener = mockk(relaxed = true)
     val gestureDetector = mockk<ShoveGestureDetector>(relaxUnitFun = true)
     presenter.addOnShoveListener(listener)
-    val result = presenter.handleShove(gestureDetector, 15.0f)
+    val result = presenter.handleShove(gestureDetector, -15.0f)
+    val expectedCameraOptions = cameraOptions {
+      pitch(2.500000022351742)
```
Comment (yunikkk) [inline plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt:863]: Why it's such a weird value of pitch?

Comment (kiryldz) [reply plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt:863]: Cause `optimizedPitch = 1.0 - (0.1 * (-15.0 + 0))` = 2.5 and simply took the value that got calculated, looks a bit strange yeah. I could potentially re-write it a bit I guess to compare doubles in a better manner.

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt (hunk 1 above)
Comment (kiryldz) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1053]: all other gestures (except move that actually gets camera options from the core) set an actual anchor

**Review by Chaoba — APPROVED** [review]

**Review by ank27 — APPROVED** [review]

**Review by yunikkk — APPROVED** [review]

**Pull request merged**